python_modules/libraries/dagster-docker/dagster_docker/container_context.py

```diff
@@ -0,0 +1,108 @@
+from typing import TYPE_CHECKING, Any, Dict, List, NamedTuple, Optional
+
+from dagster import check
+from dagster.config.validate import process_config
+from dagster.core.errors import DagsterInvalidConfigError
+from dagster.core.storage.pipeline_run import PipelineRun
+
+from .utils import DOCKER_CONTAINER_CONTEXT_SCHEMA
+
+if TYPE_CHECKING:
+    from . import DockerRunLauncher
+
+
+class DockerContainerContext(
+    NamedTuple(
+        "_DockerContainerContext",
+        [
+            ("registry", Optional[Dict[str, str]]),
+            ("env_vars", List[str]),
+            ("networks", List[str]),
+            ("container_kwargs", Optional[Dict[str, Any]]),
+        ],
+    )
+):
+    def __new__(
+        cls,
+        registry: Optional[Dict[str, str]] = None,
+        env_vars: Optional[List[str]] = None,
+        networks: Optional[List[str]] = None,
+        container_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        return super(DockerContainerContext, cls).__new__(
+            cls,
+            registry=check.dict_param(registry, "registry") if registry != None else None,
+            env_vars=check.opt_list_param(env_vars, "env_vars", of_type=str),
+            networks=check.opt_list_param(networks, "networks", of_type=str),
+            container_kwargs=(
+                check.dict_param(container_kwargs, "container_kwargs")
+                if container_kwargs != None
+                else None
+            ),
+        )
+
+    def merge(self, other: "DockerContainerContext"):
+        return DockerContainerContext(
+            registry=other.registry if other.registry != None else self.registry,
+            env_vars=self.env_vars + other.env_vars,
+            networks=self.networks + other.networks,
+            container_kwargs=other.container_kwargs
+            if other.container_kwargs != None
+            else self.container_kwargs,
```

Comment:
this part merits some discussion probably. Note how some fields (the lists, basically) are appended, while others are replaced. This mirrors the existing logic for transferring run launcher config onto executors (and is very likely what you want for things like secret) - but might be confusing.